configdir/parser.py

```diff
@@ -64,14 +68,18 @@ def parse(directory):
         raise ConfigDirMissingError("`{}` is not a directory.".format(directory))
 
     interpolate_keys = set()
-    config = _get_config_values(directory, interpolate_keys, ())
+    key_names_to_be_encoded = set()
+    config = _get_config_values(directory, interpolate_keys, key_names_to_be_encoded, ())
     interpolate = Interpolator(config)
 
     for key_names in interpolate_keys:
         c = config
         for k in key_names[:-1]:
             c = c[k]
 
-        c[key_names[-1]] = interpolate(c[key_names[-1]])
+        if key_names[-1] in key_names_to_be_encoded:
+            c[key_names[-1]] = quote(interpolate(c[key_names[-1]]), safe="")
```

Comment:
This will quote the whole URL. That will not work as a URL.

Only the interpolated parameters should be encoded. Quoting should happen in `Interpolator._get_value` and `__call__`. This should be something like:

```python
    def __call__(self, value, formatter=None):
        """
        Interpolate a value

        Args:
             value (mixed)
             formatter (callable): A optional function that takes the current key and value
                 and dictionary of values and returns the value, possibly with modifications. 
                 Example::
                     def format(key, value, values):
                         # optionally modify value
                         return value

        Returns:
            mixed: The interpolated value
        """
        if isinstance(value, Mapping):
            return {k: self(v, formatter=formatter) for k, v in value.items()}
        elif isinstance(value, (list, GeneratorType)):
            return [self(v, formatter=formatter) for v in value]
        elif isinstance(value, string_types):
            return self._key_regex.sub(partial(self._get_value, formatter=formatter), value)
        return value

    def _get_value(self, match, formatter=None):
        # etc
        if formatter:
            return formatter(full_key, value, self.values)
        return value

```

Use it like this:

```python
def _uri_value_formatter(key, value, values):
    return quote(value)

c[key_names[-1]] = interpolate(c[key_names[-1]], formatter=_uri_value_formatter)
```